.github/scripts/check-template-and-add-labels.ts

```diff
@@ -384,3 +400,37 @@ async function userBelongsToMetaMaskOrg(
 
   return Boolean(userBelongsToMetaMaskOrgResult?.user?.organization?.id);
 }
+
+// This function checks if the PR description has a changelog entry
+function hasChangelogEntry(body: string): boolean {
+  // Remove HTML comments (including multiline)
+  const uncommentedBody = body.replace(/<!--[\s\S]*?-->/g, "");
+
+  // Split body into lines
+  const lines = uncommentedBody.split(/\r?\n/);
+
+  // Find the line starting with "CHANGELOG entry:"
+  const changelogLine = lines.find(line => line.trim().startsWith("CHANGELOG entry:"));
+
+  if (!changelogLine) {
+    console.log("Changelog entry line missing");
+    return false;
+  }
+
+  // Extract everything after the prefix, tolerating extra spaces after the colon
+  const match = changelogLine.match(/^\s*CHANGELOG entry:\s*(.*)$/);
+  const entry = match?.[1]?.trim() ?? "";
+
+  if (entry === "") {
+    console.log("Changelog entry is empty");
+    return false;
+  }
+
+  if (entry.toLowerCase() === "undefined") {
```

Comment:
small question: is this check really needed? I'm wondering under which circumstances we might encounter this? 
:thinking: